google/auth/impersonated_credentials.py

```diff
@@ -227,6 +229,9 @@ def __init__(
         self.expiry = _helpers.utcnow()
         self._quota_project_id = quota_project_id
         self._iam_endpoint_override = iam_endpoint_override
+        
+        if (self._iam_endpoint_override == None):
+            self._iam_endpoint_override = iam._IAM_ENDPOINT.format(self.universe_domain, self._target_principal)
```

Comment:
Why do we have to create endpoint override always? What if target_principal is none?

Reply:
I think if override is not given by user, we should not do it. We should fallback to building the url the same way we are doing now, but with universe domain.

Reply:
This does not have to be a user-provided override. GCloud crates impersonated_credentials with iam_override without user explicitly asking for it. Here we following that path. 

We can do it differently but it will make code more complex

Reply:
If target principal in None - it is a misconfiguration that will later fail. Nothing new for the change.